### PR TITLE
Add point + range events feature

### PR DESCRIPTION
**Pull request by romankaravia**
Implements [this feature](https://3.basecamp.com/3500782/buckets/14353819/todos/2196560855).

Deployed on staging.

- [ ] Add some more fixtures and publish them on test + staging.

### CODE REVIEW

File: helpers/events.js [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+  {
+    name: "events-range-from-to",
+    type: "rule",
+    from: { data: "events-range" },
+    encode: {
+      enter: {
+        x: { signal: "floor(scale('xScale', datum.dateFrom)) + 0.5" },
+        x2: { signal: "floor(scale('xScale', datum.dateTo)) + 0.5" },
+        y: { value: -17.5 },
+        stroke: { value: "#6e6e7e" },
```
Comment (fuenkchen) [inline helpers/events.js]: I wouldn't hard code colors here and use sophie colors instead, e.g. you could use `spec.config.axis.labelColor` or you could create a new config entry for events and ranges. The latter could be also used for `font-weight` config.
(Example: https://github.com/nzzdev/Q-chart/blob/dev/chartTypes/arrow/mapping.js#L220)

File: helpers/events.js [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+    type: "text",
+    from: { data: "events-range-from-to" },
+    encode: {
+      enter: {
+        x: { signal: "(datum.x + datum.x2) / 2" },
+        y: { value: -17 },
+        align: { value: "center" },
+        baseline: { value: "middle" },
+        fill: { value: "#6e6e7e" },
+        fontWeight: { value: "bold" },
```
Comment (fuenkchen) [inline helpers/events.js]: please see the other comment on style config

File: chartTypes/commonMappings.js [hunk 3]
```diff
@@ -453,6 +448,74 @@ function getBarAxisPositioningMappings() {
...
+            name: "events-point",
+            values: pointEventValues
+          },
+          {
+            name: "events-range",
+            values: rangeEventValues
+          }
+        );
+
+        spec.marks.push(...eventHelpers.verticalMarks);
```
[the remark this answers is not in the document]
Comment (romankaravia) [reply chartTypes/commonMappings.js]: Oops, good catch => fixed

File: helpers/events.js (hunk 1 above)
Comment (romankaravia) [reply helpers/events.js]: Ok, I added a few config variables :-)

**Comment by fuenkchen** [comment]
Another little thing which came to my mind: I'm not sure if "annotation" (aka the text of event/range) should be a required field. Often we are quite cautious with implementing required fields, e.g. you could add a point to a map without labelling it.

**Comment by romankaravia** [comment]
@fuenkchen Thank you for the reviews!

> Is there a simple way to put all numbers of events and ranges within the chart in front of annotation lines, so that this does not happen

Probably not very difficult, but I think overlapping ranges and single events would still look confusing, let's discuss again next week.

> Often we are quite cautious with implementing required fields, e.g. you could add a point to a map without labelling it.

Hmmm, good point, but I'm not sure how this can work with the numbering system, for the map it works well if points on the map are annotated inline, then you just have a point without a label, but with the option "Show labels below map" it looks like this:
image

**Pull request merged**